nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -69,12 +92,140 @@ static NFCSTATUS phNciNfc_Init(void *pContext)
 static NFCSTATUS phNciNfc_ProcessInitRsp(void *pContext, NFCSTATUS Status)
 {
     NFCSTATUS wStatus = Status;
+    pphNciNfc_Context_t pNciContext = pContext;
+
+    PH_LOG_NCI_FUNC_ENTRY();
+    if (NULL != pNciContext)
+    {
+        if ((pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) <=
+            (PH_NCI1xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK))
+            wStatus = phNciNfc_ProcessInitRspNci1x(pContext, Status);
+        else if ((pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) ==
+            (PH_NCI2xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK))
+            wStatus = phNciNfc_ProcessInitRspNci2x(pContext, Status);
+    }
+    PH_LOG_NCI_FUNC_EXIT();
+    return wStatus;
+}
+
+static NFCSTATUS phNciNfc_ProcessInitRspNci2x(void *pContext, NFCSTATUS Status)
+{
+    NFCSTATUS wStatus = Status;
     phNciNfc_sInitRspParams_t *pInitRsp = NULL;
     pphNciNfc_Context_t pNciContext = pContext;
     uint8_t Offset = 0;
     uint16_t wTemp = 0;
 
     PH_LOG_NCI_FUNC_ENTRY();
+    if (NULL != pNciContext)
+    {
+        pInitRsp = &(pNciContext->InitRspParams);
+        if (pNciContext->RspBuffInfo.wLen >= 19)
```

Comment:
ok. I have defined 19 as a minimal consistent CORE_INIT_RSP Len. However, in theory this length could be 17 (without any Rf interface) but the NFC controller will not have any function.